utilities/imgcat

```diff
@@ -118,40 +159,67 @@ while [ $# -gt 0 ]; do
     -p | --p | --print)
         print_filename=1
         ;;
+    -q | --q | --quiet)
+        print_filename=0
+        ;;
+    -W | --W | --width)
+        validate_size_unit "$2"
+        width="$2"
+        shift
+        ;;
+    -H | --H | --height)
+        validate_size_unit "$2"
+        height="$2"
+        shift
+        ;;
+    -r | --R | --preserve-aspect-ratio)
+        preserve_aspect_ratio=1
+        ;;
+    -s | --S | --stretch)
+        preserve_aspect_ratio=0
+        ;;
+    -f | --F | --file)
+        has_stdin=f
+        is_url=f
+        ;;
     -u | --u | --url)
         check_dependency curl
-        encoded_image=$(curl -s "$2" | b64_encode) || (
-            error "No such file or url $2"
-            exit 2
-        )
         has_stdin=f
-        print_image "$2" 1 "$encoded_image" "$print_filename"
-        set -- "${@:1:1}" "-u" "${@:3}"
-        if [ "$#" -eq 2 ]; then
-            exit
-        fi
+        is_url=t
         ;;
     -*)
         error "Unknown option flag: $1"
         show_help
         exit 1
         ;;
     *)
-        if [ -r "$1" ]; then
-            has_stdin=f
-            print_image "$1" 1 "$(b64_encode <"$1")" "$print_filename"
+        if [ "$is_url" == "t" ]; then
+            encoded_image=$(curl -fs "$1" | b64_encode) || {
+                error "Could not retrieve image from URL $1, error_code: $?"
+                exit 2
+            }
+        elif [ -r "$1" ]; then
+            encoded_image=$(cat "$1" | b64_encode)
         else
             error "imgcat: $1: No such file or directory"
             exit 2
         fi
+        has_stdin=f
+        print_image "$1" 1 "$encoded_image" "$print_filename" "$width" "$height" "$preserve_aspect_ratio"
         ;;
     esac
     shift
 done
 
 # Read and print stdin
 if [ $has_stdin = t ]; then
-    print_image "" 1 "$(cat | b64_encode)" ""
+    print_image "" 1 "$(cat | b64_encode)" 0 "$width" "$height" "$preserve_aspect_ratio"
+fi
+
+if [ "$has_image_displayed" != "t" ]; then
```

Comment:
There should be a way to suppress this error (either `-q` or the absence of `-v`)

Reply:
This error rises only when command line arguments are invalid, i.e. no STDIN or any file / url was provided. So if this is the case, it displays help and exit with faulure code. I can add option to suppress any error messages, but why we should suppress this message? It is common and reasonable behavior in most utilities: if command-line arguments are invalid, display error and help message.

Reply:
It could happen if a glob expansion fails, in which case you'd want it silent.